ironfish-cli/src/commands/chain/download.ts

```diff
@@ -149,13 +149,20 @@ export default class Download extends IronfishCommand {
       })
 
       await new Promise<void>((resolve, reject) => {
-        writer.on('error', (e) => {
+        const onWriterError = (e: unknown) => {
+          writer.removeListener('close', onWriterClose)
+          writer.removeListener('onWriterError', onWriterError)
           reject(e)
-        })
+        }
 
-        writer.on('close', () => {
+        const onWriterClose = () => {
+          writer.removeListener('close', onWriterClose)
+          writer.removeListener('onWriterError', onWriterError)
           resolve()
```

Comment:
Would we be better off by removing `onWriterClose` altogether and writing 
```
response.data.on('end', () => {
  writer.close()
  resolve()
}
```
?

Reply:
I think so, yeah.